src/main/java/io/cdap/plugin/snowflake/common/client/SnowflakeAccessor.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.snowflake.common.client;
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.dataset.lib.KeyValue;
+import io.cdap.plugin.common.KeyValueListParser;
+import io.cdap.plugin.snowflake.common.BaseSnowflakeConfig;
+import io.cdap.plugin.snowflake.common.OAuthUtil;
+import io.cdap.plugin.snowflake.common.exception.ConnectionTimeoutException;
+import io.cdap.plugin.snowflake.common.util.QueryUtil;
+import net.snowflake.client.jdbc.SnowflakeBasicDataSource;
+import org.apache.http.impl.client.HttpClients;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * A class which accesses Snowflake API.
+ */
+public class SnowflakeAccessor {
+  private static final String APPLICATION_NAME = "CDAP";
+  private static final int LIMIT_ROWS = 5;
+
+  private final BaseSnowflakeConfig config;
+  protected final SnowflakeBasicDataSource dataSource;
+
+  public SnowflakeAccessor(BaseSnowflakeConfig config) {
+    this.config = config;
+    this.dataSource = new SnowflakeBasicDataSource();
+    initDataSource(dataSource, config);
+  }
+
+  public void runSQL(String query) throws IOException {
+    try (Connection connection = dataSource.getConnection();
+         PreparedStatement populateStmt = connection.prepareStatement(query);) {
+      populateStmt.execute();
+    } catch (SQLException e) {
+      throw new IOException(String.format("Statement '%s' failed.", query), e);
+    }
+
+  }
+
+  private static final Logger LOG = LoggerFactory.getLogger(SnowflakeAccessor.class);
+
+  /**
+   * Returns field descriptors for specified import query.
+   *
+   * @return List of field descriptors.
+   * @throws IOException thrown if there are any issue with the I/O operations.
+   */
+  public List<SnowflakeFieldDescriptor> describeQuery(String query) throws IOException {
+    String importQuery = QueryUtil.limitQuery(query, LIMIT_ROWS);
```

Comment:
why do we set LIMIT_ROWS = 5? why not, say, 1?

Reply:
Set to 1